pkg/clients/has/components.go

```diff
@@ -182,13 +182,23 @@ func (h *HasController) WaitForComponentPipelineToBeFinished(component *appservi
 	return nil
 }
 
-// Universal method to create a component in the kubernetes clusters.
-func (h *HasController) CreateComponent(componentSpec appservice.ComponentSpec, namespace string, outputContainerImage string, secret string, applicationName string, skipInitialChecks bool, annotations map[string]string) (*appservice.Component, error) {
+// Universal method to create a component in the kubernetes clusters and adds a suffix to the component name to allow multiple components with unique names.
+// Generate a random component name with #combinations > 11M, Create unique resource names that adhere to RFC 1123 Label Names
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+func (h *HasController) CreateComponentV2(componentSpec appservice.ComponentSpec, namespace string, outputContainerImage string, secret string, applicationName string, skipInitialChecks bool, annotations map[string]string, option string) (*appservice.Component, string, error) {
```

Comment:
Is it possible to have a default value for function parameter if not provided in Go? If so, maybe you do not need to create `CreateComponentV2()` at all.

Reply:
OK, looks like it is not possible in Go.

Reply:
CreateComponentV2 is a wrapper function for CreateComponent and componentName is not given as an input parameter in the original CreateComponent so for backwards compatibility we use that method and to my understanding we do need CreateComponentV2